.github/workflows/check.yaml

```diff
@@ -0,0 +1,33 @@
+name: Analyze code and run tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: ⬇️ Checkout repository
+        uses: actions/checkout@v3
+
+      - name: ⚙️ Setup Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          flutter-version: "3.16.3"
```

Comment:
I think we could move it to GitHub variables in case we need to use it in another workflow. Also, it will be easier to edit the Flutter version later on (without changing the workflow file).

Reply:
Also, not sure if we should run it against Flutter as it is a pure dart package.

Reply:
Good point, it's pure dart now 😁 https://github.com/nomtek/mistralai_client_dart/pull/14/commits/111b38b0e14b62345ed6141db6619876ed8e8527